### ISSUE TITLE
Need High Availability (HA) Guides with reference architecture diagrams.

### ISSUE THREAD

**Issue by Valien** [gravitational/teleport#16751]
## Details
We talk about HA all the time with customers and in those conversations we reference the need for ALB/NLB, etc for HA in non-Kubernetes environments. Our docs do not have any guides whatsoever that covers HA setup for non-K8s environments.

Here is what we currently have in our docs with regards to HA:

- https://goteleport.com/docs/deploy-a-cluster/deployments/aws-terraform/
- https://goteleport.com/docs/database-access/guides/ha/
- https://goteleport.com/docs/deploy-a-cluster/teleport-cloud/architecture/#high-availability
- https://goteleport.com/docs/deploy-a-cluster/helm-deployments/gcp/
- https://goteleport.com/docs/deploy-a-cluster/helm-deployments/aws/
- https://goteleport.com/docs/deploy-a-cluster/deployments/ibm/#ibm-teleport-introduction — only page with some basic arch diagrams showing auto-scaling groups
- https://goteleport.com/docs/reference/backends/

And this basic sentence here:

> "You can run multiple auth services in the cluster for high availability." https://goteleport.com/docs/architecture/overview/

Basically only HA info listed in our docs is related to K8s and the like. Nothing for non-K8s HA (besides some DB info). No diagrams or reference architecture showing LB’s, etc.

There are some internal reference guides that our CSM teams use that we reference all the time but these are internal facing documents and not available on our public docs (even generic versions). 

There are some related issues: #5050 , #3760, #5093

### Category

- New Feature
- Improve Existing

**Comment by ptgott** [comment]
I think we should also document if there's a single canonical HA approach, or if there are a few different ones that work for different use cases. We can add a guide that describes one or more architectural patterns that enable scalability in a Teleport cluster.

In the architectural guide to sclability, l we can also describe the ports/communication flows involved in a scalable Teleport cluster (see #5619).

**Comment by ptgott** [comment]
We can consider adding the architectural guide to scaling to our [current scaling guide](https://github.com/gravitational/teleport/blob/237da44f6ca0d52f3e2ca279974d050b20bb2ca9/docs/pages/management/operations/scaling.mdx).
